Writing logs files failed after writing a lot of log lines

I've got a process that outputs a lot of log lines, of which 33% is actual JSON. After a minute the output breaks with the following error:

```
panic: JSON decoder out of sync - data changing underfoot?
goroutine 1 [running]:
encoding/json.(*decodeState).objectInterface(0xc000263d60, 0xc000395980)
	/usr/local/Cellar/go/1.12.4/libexec/src/encoding/json/decode.go:1084 +0x2cd
encoding/json.(*decodeState).object(0xc000263d60, 0x11713c0, 0xc000395980, 0x16, 0xc000263d88, 0xc00037537b)
	/usr/local/Cellar/go/1.12.4/libexec/src/encoding/json/decode.go:623 +0x215c
encoding/json.(*decodeState).value(0xc000263d60, 0x11713c0, 0xc000395980, 0x16, 0xc0000e7e08, 0x1114da4)
	/usr/local/Cellar/go/1.12.4/libexec/src/encoding/json/decode.go:381 +0x6e
encoding/json.(*decodeState).unmarshal(0xc000263d60, 0x11713c0, 0xc000395980, 0xc000263d88, 0x0)
	/usr/local/Cellar/go/1.12.4/libexec/src/encoding/json/decode.go:179 +0x209
encoding/json.Unmarshal(0xc0000c7eeb, 0xd7, 0x115, 0x11713c0, 0xc000395980, 0xc0000c7ac7, 0x0)
	/usr/local/Cellar/go/1.12.4/libexec/src/encoding/json/decode.go:106 +0x123
main.main()
	/Users/jan-willemvandermeer/projects/go/src/github.com/koenbollen/jl/main.go:42 +0x5d3
log writing failed. Broken pipe @ io_write - <STDOUT>
log writing failed. Broken pipe @ io_write - <STDOUT>
log writing failed. Broken pipe @ io_write - <STDOUT>
```

--- a/stream/logs.go
+++ b/stream/logs.go
@@ -51,10 +51,14 @@ func (l *stream) run() {
 		json := l.parse(raw)
 		prefix, suffix := split(raw, json)
 		line := &Line{
-			Raw:    raw,
-			JSON:   json,
+			Raw:    make([]byte, len(raw)),
 			Prefix: prefix,
 			Suffix: suffix,
+		}
+		copy(line.Raw, raw)
+		if json != nil {
+			line.JSON = make([]byte, len(json))
+			copy(line.JSON, json)
 		}
 		select {
 		case <-l.stop:

